client/core/core_test.go

```diff
@@ -2518,15 +2518,17 @@ func TestHandlePreimageRequest(t *testing.T) {
 		t.Fatal("no order note from preimage request handling")
 	}
 
-	// Note, even though tracker.csum is initialized in a separate go routine (in handlePreimageRequest),
-	// we have it here properly synchronized due to using channels above ^.
+	// Since tracker might be modified in different go-routines from the one running here,
+	// we need to properly synchronize any read for its fields.
+	tracker.mtx.RLock()
 	if !bytes.Equal(commitCSum, tracker.csum) {
 		t.Fatalf(
 			"handlePreimageRequest must initialize tracker csum, exp: %s, got: %s",
 			commitCSum,
 			tracker.csum,
 		)
 	}
+	tracker.mtx.RUnlock()
```

Comment:
Deadlock if it does hit t.Fatal.

Reply:
true, although I believe `TestHandlePreimageRequest` to be a pretty simple isolated test case with nothing vital here being locked forever,

when we hit `t.Fatal` here, all the related stuff around `tracker.mtx` becomes useless,

but maybe it is better to just copy value, re-wrote this part

Reply:
I get what you're saying, but it could hang at 

```
defer rig.shutdown()
````

`Core` might be unable to stop depending on what it's doing at the time.  I've had tests hang on t.Fatals before and it's very confusing so best to play it safe.  Thanks for changing.